src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -315,22 +299,47 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
   }
 
   @Override
-  public ResponseEntity<JobModel> createGoogleContext(
-      UUID id, @Valid CreateGoogleContextRequestBody body) {
+  public ResponseEntity<CreateCloudContextResult> createCloudContext(
+      UUID id, @Valid CreateCloudContextRequest body) {
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
-    // TODO(PF-153): Use the optional jobId from the body for idempotency instead of always creating
-    // a new job id.
-    String jobId = workspaceService.createGoogleContext(id, userReq);
-    JobModel jobModel = jobService.retrieveJob(jobId, userReq);
-    // TODO(PF-221): Fix the jobs polling location once it exists.
-    return ResponseEntity.status(HttpStatus.ACCEPTED)
-        .location(URI.create(String.format("/api/jobs/v1/%s", jobId)))
-        .body(jobModel);
+    ControllerValidationUtils.validateCloudContext(body.getCloudContext());
+    String jobId = body.getJobControl().getId();
+    String resultPath = getAsyncResultEndpoint(jobId);
+
+    workspaceService.createGoogleContext(id, jobId, resultPath, userReq);
+    CreateCloudContextResult response = fetchCreateCloudContextResult(jobId, userReq);
+    return new ResponseEntity<>(
+        response, HttpStatus.valueOf(response.getJobReport().getStatusCode()));
+  }
+
+  @Override
+  public ResponseEntity<CreateCloudContextResult> createCloudContextResult(UUID id, String jobId) {
```

Comment:
This endpoint does return the result of a `createCloudContext` call if the job is finished (in `CreateCloudContextResult.getGoogleContext()`), though. 
I think the pattern of naming an async endpoint `doFoo` and its result polling endpoint `doFooResult` matches our model of async APIs well, though yeah it can be a little weird to parse.

Reply:
Can we compromise on `getCreateCloudContextResult()`?